mne/export/_edf.py

```diff
@@ -0,0 +1,184 @@
+        sex = subj_info.get('sex')
+
+        if birthday is not None:
+            if hdl.setPatientBirthDate(birthday[0], birthday[1], birthday[2]) != 0:  # noqa
+                raise RuntimeError("setPatientBirthDate() returned an error")
+        if hdl.setPatientName(name) != 0:  # noqa
+            raise RuntimeError("setPatientName() returned an error")
+        if hdl.setPatientGender(sex) != 0:  # noqa
+            raise RuntimeError("setPatientGender() returned an error")
+
+        if hdl.setAdditionalPatientInfo(f"hand={hand}") != 0:  # noqa
+            raise RuntimeError("setAdditionalPatientInfo() returned an error")
+
+    # set measurement date
+    meas_date = raw.info['meas_date']
+    if meas_date:
+        subsecond = meas_date.microsecond / 100.
+        if hdl.setStartDateTime(year=meas_date.year, month=meas_date.month,
+                                day=meas_date.day, hour=meas_date.hour,
+                                minute=meas_date.minute,
+                                second=meas_date.second,
+                                subsecond=subsecond) != 0:  # noqa
+            raise RuntimeError("setStartDateTime() returned an error")
+    # if hdl.setAdministrationCode("1234567890") != 0:
+    #     raise RuntimeError("setAdministrationCode() returned an error")
+    #     sys.exit()
+    # if hdl.setTechnician("Black Jack") != 0:
+    #     raise RuntimeError("setTechnician() returned an error")
+    #     sys.exit()
+
+    device_info = raw.info.get('device_info')
+    if device_info is not None:
+        device_type = device_info.get('type')
+        if hdl.setEquipment(device_type) != 0:  # noqa
+            raise RuntimeError("setEquipment() returned an error")
+    # if hdl.setAdditionalRecordingInfo("nothing special") != 0:
+    #     raise RuntimeError("setAdditionalRecordingInfo() returned an error")
+    #     sys.exit()
+
+    # Write each second (i.e. datarecord) separately.
+    for isec in range(np.ceil(n_secs).astype(int)):
+        end_samp = (isec + 1) * sfreq
+        if end_samp > n_times:
+            end_samp = n_times
+        start_samp = isec * sfreq
+
+        # then for each second write each channel
+        for ich in range(n_chs):
+            # create a buffer with sampling rate
+            buf = np.zeros(sfreq, np.float64, "C")
+
+            # get channel data for this second
+            ch_data = data[ich, start_samp:end_samp]
+
+            buf[:len(ch_data)] = ch_data
+            err = hdl.writeSamples(buf)
+            if err != 0:  # noqa
+                raise RuntimeError(f"writeSamples() returned error: {err}")
+
+        # there was an incomplete datarecord
+        if len(ch_data) != len(buf):
+            warn(f'A complete data record consists of {len(buf)} samples, '
+                 f'but this sample window ended up having {len(ch_data)} '
+                 f'samples. {len(buf) - len(ch_data)} zeros were appended '
+                 f'to the datarecord.')
```

Comment:
end-users don't need to know the block size, they just need to know what happened and why:

"EDF format requires equal-length data blocks, so XXX seconds of zeros were appended to all channels when writing the final block."

I'm not sure whether expressing XXX in (fractional) seconds or samples is more useful, WDYT?

Reply:
I think fractional seconds because this wouldn't occur unless there is a "not-full" record (i.e. full cycle of a sampling rate).

Adjusted the warning to be what you recommended.